### PR TITLE
Sprint c02   get announcement   2

### PR DESCRIPTION
**Pull request by DDanielVieira**

### CODE REVIEW

File: src/main/java/com/criticalsoftware/Announcement.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.time.LocalDateTime;
+
+public class Announcement extends PanacheMongoEntity {
+
+    private ObjectId id;
+    private Product product;
+    private LocalDateTime date;
+    private ObjectId userDonor;
+
+    public Announcement() {
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Announcement.java]: See messages on Martin's PR, please.

File: src/main/java/com/criticalsoftware/Product.java [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
 public class Product {
 
     private ObjectId id;
+    private String name;
     private String description;
-    private BufferedImage photo;
+    private String photo;
     private String category;
-    private User user;
+    private User userDonnor;
```
Comment (jpmoreira-training) [inline src/main/java/com/criticalsoftware/Product.java]: Does product need a userDonnor? Who has a userdonnor? The product? The Announcement?

**Review by jpmoreira-training — CHANGES_REQUESTED** [review]

File: src/main/java/com/criticalsoftware/Product.java (hunk 2 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/Product.java]: Understood and done.

File: src/main/java/com/criticalsoftware/Announcement.java (hunk 1 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/Announcement.java]: Ok, seen and done.

**Review by jrromao — CHANGES_REQUESTED** [review]
You can use lombok to avoid the explicit declaration of the constructors, getters and setters.

File: src/main/java/com/criticalsoftware/Product.java [hunk 3]
```diff
@@ -1,23 +1,32 @@
 package com.criticalsoftware;
+
 import org.bson.types.ObjectId;
-import java.awt.image.BufferedImage;
 
 public class Product {
 
     private ObjectId id;
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Product.java:7]: As the id field is already inherited from the parent class (PanacheMongoEntity), it doesn't need to be explicitly declared in this class.

File: src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    private String id;
+    private Product product;
+    private LocalDateTime date;
+    private ObjectId userDonorId;
+    private ObjectId userDoneeId;
+
+
+    public AnnouncementResponse(String id, Product product, ObjectId userDonorId,ObjectId userDoneeId) {
+
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/giveaway/AnnouncementResponse.java:18]: Why do you need this empty constructor?

File: src/main/java/com/criticalsoftware/giveaway/Announcement.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    public void setProduct(Product product) {
+        this.product = product;
+    }
+
+    //Date
+    public LocalDateTime getdate(){
+        return this.date;
+    }
+
+    public LocalDateTime getLocalDateTime() {
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/giveaway/Announcement.java:52]: You should use getDate() and remove this getLocalDateTime().

File: src/main/java/com/criticalsoftware/giveaway/Announcement.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    public Product getProduct(){
+        return this.product;
+    }
+
+    public void setProduct(Product product) {
+        this.product = product;
+    }
+
+    //Date
+    public LocalDateTime getdate(){
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/giveaway/Announcement.java:48]: Method names should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

File: src/main/java/com/criticalsoftware/Product.java [hunk 7]
```diff
@@ -1,23 +1,32 @@
 package com.criticalsoftware;
+
 import org.bson.types.ObjectId;
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Product.java:3]: You can remove this unused import.

File: src/main/java/com/criticalsoftware/Product.java [hunk 8]
```diff
@@ -1,23 +1,32 @@
...
-    private BufferedImage photo;
+    private String photo;
     private String category;
-    private User user;
 
     public Product() {
     }
 
-    public Product(String description, BufferedImage photo, String category, User user) {
+    public Product(String name,String description, String photo, String category, User userDonnor) {
```
Comment (jrromao) [inline src/main/java/com/criticalsoftware/Product.java:16]: Why do you have the input parameter "userDonnor" if you don't use it in the constructor?

**Review by jrromao — CHANGES_REQUESTED** [review]
You can use lombok to avoid the explicit declaration of the constructors, getters and setters.

File: src/main/java/com/criticalsoftware/Product.java (hunk 3 above)
Comment (DDanielVieira) [reply src/main/java/com/criticalsoftware/Product.java:7]: Done.

**Pull request merged**